jerry-core/parser/js/js-scanner.c

```diff
@@ -0,0 +1,1702 @@
+          {
+            scanner_context.mode = (stack_top == SCAN_STACK_BLOCK_EXPRESSION ? SCAN_MODE_PRIMARY_EXPRESSION_END
+                                                                             : SCAN_MODE_STATEMENT);
+            parser_stack_pop_uint8 (context_p);
+            break;
+          }
+
+          if (lexer_compare_literal_to_identifier (context_p, "static", 6))
+          {
+            lexer_next_token (context_p);
+          }
+
+          if (lexer_compare_literal_to_identifier (context_p, "get", 3)
+              || lexer_compare_literal_to_identifier (context_p, "set", 3))
+          {
+            lexer_next_token (context_p);
+          }
+
+          parser_stack_push_uint8 (context_p, SCAN_STACK_CLASS);
+
+#if ENABLED (JERRY_ES2015_OBJECT_INITIALIZER)
+          if (context_p->token.type == LEXER_LEFT_SQUARE)
+          {
+            parser_stack_push_uint8 (context_p, SCAN_STACK_COMPUTED_PROPERTY);
+            scanner_context.mode = SCAN_MODE_PRIMARY_EXPRESSION;
+            break;
+          }
+#endif /* ENABLED (JERRY_ES2015_OBJECT_INITIALIZER) */
+
+          scanner_context.mode = SCAN_MODE_FUNCTION_ARGUMENTS;
+          continue;
+        }
+#endif /* ENABLED (JERRY_ES2015_CLASS) */
+#if ENABLED (JERRY_ES2015_ARROW_FUNCTION)
+        case SCAN_MODE_ARROW_FUNCTION:
+        {
+          if (type == LEXER_ARROW)
+          {
+            lexer_next_token (context_p);
+
+            if (context_p->token.type == LEXER_LEFT_BRACE)
+            {
+              parser_stack_push_uint8 (context_p, SCAN_STACK_BLOCK_EXPRESSION);
+              scanner_context.mode = SCAN_MODE_STATEMENT;
+            }
+            else
+            {
+              scanner_context.mode = SCAN_MODE_PRIMARY_EXPRESSION;
+              continue;
+            }
+            break;
+          }
+          scanner_context.mode = SCAN_MODE_POST_PRIMARY_EXPRESSION;
+          /* FALLTHRU */
+        }
+#endif /* ENABLED (JERRY_ES2015_ARROW_FUNCTION) */
+        case SCAN_MODE_POST_PRIMARY_EXPRESSION:
+        {
+          if (scanner_scan_post_primary_expression (context_p, &scanner_context, type))
+          {
+            break;
+          }
+          /* FALLTHRU */
+        }
+        case SCAN_MODE_PRIMARY_EXPRESSION_END:
+        {
+          if (scanner_scan_primary_expression_end (context_p, &scanner_context, type, stack_top))
+          {
+            continue;
+          }
+          break;
+        }
+        case SCAN_MODE_STATEMENT:
+        {
+          if (scanner_scan_statement (context_p, &scanner_context, type, stack_top))
+          {
+            continue;
+          }
+          break;
+        }
+        case SCAN_MODE_FUNCTION_ARGUMENTS:
+        {
+#if ENABLED (JERRY_ES2015_CLASS)
+          JERRY_ASSERT (stack_top == SCAN_STACK_BLOCK_STATEMENT
+                        || stack_top == SCAN_STACK_BLOCK_EXPRESSION
+                        || stack_top == SCAN_STACK_CLASS
```

Comment:
How about only a single assert with a guarded `stack_top_p == SCAN_STACK_CLASS` ?

Reply:
Because ASSERT is a macro, you cannot put guard into it. Some compilers support it but it is non-standard.